data/transaction/apiTransactionResult.go

```diff
@@ -59,6 +59,12 @@ type ApiTransactionResult struct {
 	InitiallyPaidFee                  string                    `json:"initiallyPaidFee,omitempty"`
 	IsRelayed                         bool                      `json:"isRelayed,omitempty"`
 	IsRefund                          bool                      `json:"isRefund,omitempty"`
+	CallType                          int                       `json:"callType,omitempty"`
```

Comment:
question for other reviewers: DirectCall for example is `0` so the `callType` field won't be included in the result. However, regular transactions won't have this value set. What should we do in such cases? 

A suggestion would be to make it a string that can have the following values: 
- `""` empty
- `directCall`
- `asynchronousCall`
- `asynchronousCallBack`
- `esdtTransferAndExecute`
- `execOnDestByCaller`

Reply:
For both situations, it's unfortunate we use zeros as constants that actually mean something 
```go
type CallType int
// Always suggested when creating something like this:
const (
  DirectCall = iota
  Async,
  ...
)
// To do it like this:
const (
  DirectCall = iota + 1
  ...
)
```

But now that we are in this situation, we have what I think is a better option for clients interpreting this result (but may be a bit weirder in implementation):
```go
type ApiTransactionResult struct {
 // This way this field will only be omitted when passing a nil pointer
 CallType *int `json:"callType,omitempty"`
}

// This will marshal to {callType: 0}
res1 := ApiTransactionResult{0}

// This will marshal to {}
res2 := ApiTransactionResult{nil}
```

Reply:
Alright, I was just exposing that there is another approach that could yield the output we want. @bogdan-rosianu 's approach would be a cleaner implementation on our side, I agree! 👍